mpas_analysis/ocean/plot_depth_integrated_time_series_subtask.py

```diff
@@ -354,6 +366,13 @@ def run_task(self):  # {{{
                                     'not be plotted.')
                 preprocessedReferenceRunName = 'None'
 
+            # rolling mean seems to have trouble with dask data sets to we
+            # write out the data set and read it back as a single-file data set
+            # (without dask)
+            dsPreprocessed = dsPreprocessed.drop('xtime')
+            write_netcdf(dsPreprocessed, self.preprocessedFileName)
+            dsPreprocessed = xarray.open_dataset(self.preprocessedFileName)
```

Comment:
@pwolfram, if you have time to take a look at this, it's mainly a question of seeing if you're good with this solution for converting a multi-file data set to a single file data set or if you'd suggest some other way of handling the issue.

Reply:
My understanding from S Hoyer is that this is the preferred way to handle these types of problems.  The only issue I foresee with this is that as data becomes larger, this doesn't scale too well.  However, recent work has been to develop parallel writing functionality into xarray so I wouldn't worry about this in the short term.

I think this is a reasonable way to convert a multi-file dataset into a single file data set 👍